Add drag-and-drop of customizer controls and have that update order in preview

westonruter opened the pull request:
The widget form controls in each sidebar customizer section could re-ordered via drag-and-drop.

Commit:
added eclipse project files to gitignore

Commit:
removed .gitignore from itself, already tracked

Commit:
removed unicode chars

Commit:
fixed plugin URLs for files that are outside the WordPress content directory (IE: symlinked)

Commit:
ignore eclipse .buildpath

Commit:
working sortables

bobbravo2 commented:
Added in pull request #14

Commit:
added nonce check

Commit:
remove outdated comment

westonruter committed:
Merge branch 'develop' of https://github.com/bobbravo2/wp-widget-customizer into bobbravo2-develop

westonruter commented:
@bobbravo2 thanks!! I did a code review of your pull request. Sorry for the barrage of comments, but I hope they're helpful! Once you process my feedback, I'll pull your commits into this issue directly (creating a `issue-1-drag-n-drop` branch), and we can then continue collaborating by my pushing my and your commits to this branch, and you can pull down commits from the branch to continue working in your fork.

Commit:
fixed camelCase and removed trailing whitespace

Commit:
cs fixes

Commit:
added static class method to get plugin url
cs fixes

Commit:
traling whitespace

bobbravo2 commented:
@westonruter  just caught your comments here. NP with the comments - I really appreciate the feedback! I'm pretty sure I've addressed everything... let me know :-D

westonruter committed:
Merge branch 'develop' of git://github.com/bobbravo2/wp-widget-customizer into bobbravo2-develop

westonruter commented:
@bobbravo2 as I commented on your PR #14, I merged in your commits into the [`issue-1-reordering`](https://github.com/x-team/wp-widget-customizer/tree/issue-1-reordering) branch, which I then converted into an internal pull request from that branch to the `develop` branch. I'll continue merging in additional commits you have from your `develop` branch.

bobbravo2 commented:
I thought you were on Vacation! I caught you on IRC at some point. Have a great Vaca, We'll catch up next week

westonruter committed:
Merge branch 'develop' into issue-1-reordering

westonruter committed:
Apply some formatting fixes

westonruter committed:
Add pointer:move to widget titles

westonruter committed:
Prevent section description from getting sorted

westonruter committed:
Add control for managing ordering of widgets in sidebar

westonruter committed:
Disambiguate nonces and ajax actions

westonruter committed:
Add @bobbravo2 to contributors

westonruter commented on widget-customizer.js:
```diff
@@ -4,14 +4,54 @@ var WidgetCustomizer = (function ($) {
+			var control = this;
+			this.container.closest( '.accordion-section-content' ).sortable({
+				items: '> .customize-control-widget_form',
+				axis: 'y',
```
@bobbravo2 note the addition of `items` which prevents the section description from being draggable, and the addition of the `axis` since it only makes sense to drag vertically.

westonruter commented:
@bobbravo2 @shaunandrews OK, I've added commits to make widget re-ordering previewable, and to lay the groundwork for adding and deleting widgets via customizer. Please check it out (this `issue-1-reordering` branch), and if it works for you, I'll release version 0.6 of the plugin.

westonruter committed:
Update ordering of widget form controls when setting is updated

Ensures that UI responds to changes to model, just as currently we update model in reponse to UI changes.
